components/script/dom/htmlimageelement.rs

```diff
@@ -432,16 +433,17 @@ impl HTMLImageElement {
             _ => {},
         };
 
-        let map = self.upcast::<Node>()
-                      .following_siblings()
+        let useMapElements =  self.upcast::<Node>()
+                      .GetRootNode()
+                      .traverse_preorder()
                       .filter_map(Root::downcast::<HTMLMapElement>)
                       .find(|n| n.upcast::<Element>().get_string_attribute(&LocalName::from("name")) == last);
 
-        let elements: Vec<Root<HTMLAreaElement>> = map.unwrap().upcast::<Node>()
-                      .children()
-                      .filter_map(Root::downcast::<HTMLAreaElement>)
-                      .collect();
-        Some(elements)
+        useMapElements.map(|mapElem| {
+            mapElem.upcast::<Node>()
+                .traverse_preorder()
+                .filter_map(Root::downcast::<HTMLAreaElement>).collect()
+        })
```

Comment:
Actually I just realised that you should rather add a method on `HTMLMapElement` itself that returns the `Vec<Root<HTMLAreaElement>>`, and call it here.